### PR TITLE
Rachel - L103.ghg_an_USA_S_T_Y

### PR DESCRIPTION
**Pull request by rhoesly**
Produces methane emission factors for animals by animals type (US, 2005)

### CODE REVIEW

File: R/zchunk_L103.ghg_an_USA_S_T_Y.R [hunk 1]
```diff
@@ -34,40 +36,69 @@ module_emissions_L103.ghg_an_USA_S_T_Y_DISABLED <- function(command, ...) {
...
-    # NOTE: there are `merge` and/or 'match' calls in this code. Be careful!
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Merge-and-Match
-    # ===================================================
+    # Map EPA ghg emissions to GCAM sector and fuel,
+    # aggregate and convert from Gg to Tg
+
+    if(OLD_DATA_SYSTEM_BEHAVIOR) {
+    # Old system yields an emissions estimate of 0 for poultry because it has NA values.
+    L103.ghg_tg_USA_an_Sepa_F_2005 <- EPA_FCCC_AG_2005 %>%
+      left_join(EPA_ghg_tech, by = "Source_Category") %>%
```
Comment (bpbond) [inline R/zchunk_L103.ghg_an_USA_S_T_Y.R]: Can the more restrictive `left_join_error_no_match` be used here?

File: R/zchunk_L103.ghg_an_USA_S_T_Y.R [hunk 2]
```diff
@@ -34,40 +36,69 @@ module_emissions_L103.ghg_an_USA_S_T_Y_DISABLED <- function(command, ...) {
...
+      mutate_each( funs(replace(., is.na(.), 0))) %>%
+      mutate_if(is.numeric, funs( .*CONV_GG_TG))
+
+    } else {
+      # Corrected poltry emissions estimate, methane emissions for poultry are relatively small
+      # compared to other animals, but non-zero
+      L103.ghg_tg_USA_an_Sepa_F_2005 <- EPA_FCCC_AG_2005 %>%
+        left_join(EPA_ghg_tech, by = "Source_Category") %>%
+        group_by(sector, fuel) %>%
+        summarize_if(is.numeric, sum, na.rm = T) %>%
```
Comment (bpbond) [inline R/zchunk_L103.ghg_an_USA_S_T_Y.R]: Please use `TRUE`

File: R/zchunk_L103.ghg_an_USA_S_T_Y.R [hunk 3]
```diff
@@ -34,40 +36,69 @@ module_emissions_L103.ghg_an_USA_S_T_Y_DISABLED <- function(command, ...) {
...
+
+    L103.ghg_tg_USA_an_Sepa_F_2005 <- EPA_FCCC_AG_2005 %>%
+      left_join(EPA_ghg_tech, by = "Source_Category") %>%
+      group_by(sector, fuel) %>%
+      summarize_if(is.numeric, sum) %>%
+      filter( !is.na(sector) , !is.na(fuel) ) %>%
+      mutate_each( funs( replace(., is.na(.), 0)) ) %>%
+      mutate_if(is.numeric, funs(.*CONV_GG_TG))
+
+    # Aggergate FAO production to EPA sectors
```
Comment (bpbond) [inline R/zchunk_L103.ghg_an_USA_S_T_Y.R]: "Aggregate"

File: R/zchunk_L103.ghg_an_USA_S_T_Y.R [hunk 4]
```diff
@@ -34,40 +36,69 @@ module_emissions_L103.ghg_an_USA_S_T_Y_DISABLED <- function(command, ...) {
...
+    L107.an_Prod_US_Sepa_2005 <- L107.an_Prod_Mt_R_C_Sys_Fd_Y %>%
+      left_join(GCAM_sector_tech,
+                by = c('GCAM_commodity' = 'sector', 'system' = 'fuel' , 'feed' = 'technology')) %>%
+      ungroup %>%
+      mutate(year = as.numeric(year)) %>%
+      filter(year == 2005, GCAM_region_ID == 1) %>%
```
Comment (bpbond) [inline R/zchunk_L103.ghg_an_USA_S_T_Y.R]: Unclear what's happening here. Could a comment explain why we're picking out USA 2005?

File: R/zchunk_L103.ghg_an_USA_S_T_Y.R [hunk 5]
```diff
@@ -34,40 +36,69 @@ module_emissions_L103.ghg_an_USA_S_T_Y_DISABLED <- function(command, ...) {
...
+      group_by(EPA_agg_sector, year) %>%
+      select(EPA_agg_sector, year,value) %>%
+      summarise_if(is.numeric, sum)
+
+    # Calculate CH4 emissions factor
+    L103.ghg_tg_USA_an_Sepa_F_2005 %>%
+      left_join_error_no_match(L107.an_Prod_US_Sepa_2005, by = c('sector' = 'EPA_agg_sector')) %>%
+      mutate(ch4_em_factor = CH4/value) %>%
+      select(sector, fuel, ch4_em_factor) %>%
+      ungroup() -> L103.ghg_tgmt_USA_an_Sepa_F_2005
```
Comment (bpbond) [inline R/zchunk_L103.ghg_an_USA_S_T_Y.R]: I prefer this (assignment at end of pipeline, as opposed to at beginning) but whichever you choose, please be consistent.

File: R/zchunk_L103.ghg_an_USA_S_T_Y.R [hunk 6]
```diff
@@ -34,40 +36,69 @@ module_emissions_L103.ghg_an_USA_S_T_Y_DISABLED <- function(command, ...) {
...
+      mutate_if(is.numeric, funs( .*CONV_GG_TG))
+
+    } else {
+      # Corrected poltry emissions estimate, methane emissions for poultry are relatively small
+      # compared to other animals, but non-zero
+      L103.ghg_tg_USA_an_Sepa_F_2005 <- EPA_FCCC_AG_2005 %>%
+        left_join(EPA_ghg_tech, by = "Source_Category") %>%
+        group_by(sector, fuel) %>%
+        summarize_if(is.numeric, sum, na.rm = T) %>%
+        filter( !is.na(sector) , !is.na( fuel )) %>%
```
Comment (bpbond) [inline R/zchunk_L103.ghg_an_USA_S_T_Y.R]: Don't use GCAM-style spacing with parentheses please.

File: R/zchunk_L103.ghg_an_USA_S_T_Y.R (hunk 1 above)
Comment (rhoesly) [inline R/zchunk_L103.ghg_an_USA_S_T_Y.R]: No, in this case there should be NAs. 
Will add comments. 
Will make corrections in other places to use `left_join_error_no_match` when applicable.

**Comment by rhoesly** [comment]
Note that for animal production methane emission factors the old data system uses US, 2005 values  for ALL historical emission factors, for all regions. There is much better data available (ours!) - historical emission factors overtime by region, but Steve recommended not opening an issue because there are plans in the future to use CEDS historical emission factors anyway.

**Comment by bpbond** [comment]
Thanks. This is now holding on review or assignment by @cahartin

**Comment by cahartin** [comment]
@rhoesly does this file inst/extdata/emissions/EPA_FCCC_AG_2005.csv  have a Source?

**Comment by rhoesly** [comment]
@cahartin The source is probably from the EPA Inventory of U.S. Greenhouse Gas Emissions and Sinks (UNFCCC submissions) but I can't find that specific table anywhere on the web anywhere.

**Comment by cahartin** [comment]
@rhoesly should we open this as an issue?

**Comment by rhoesly** [comment]
issue added! @cahartin

**Comment by cahartin** [comment]
@bpbond can we approve this or should we track down the source of the emissions?

**Comment by bpbond** [comment]
Fine by me, as long as we have this tracked in a separate issue. I'll merge unless you tell me otherwise.

**Pull request merged**